src/smc-webapp/project_files.cjsx

```diff
@@ -1987,14 +1987,17 @@ exports.ProjectFiles = rclass ({name}) ->
                 style   = {error_style}
                 onClose = {=>@props.actions.setState(error:'')} />
 
+    render_error_message: ->
+        <div>You are not logged in. To view non-public files or edit files in this project you{"'"}ll need to <Button bsStyle={'success'} onClick={=>$('.sign_in *').trigger('click')}>sign in</Button>. If you are not a collaborator then you need to ask a collaborator of the project to add you to access non public files.</div>
```

Comment:
@haraldschilly any possibility of OKing this? I did this because it seemed non-trival for me to figure out how to do the action otherwise. Like I would have to move a bunch of code around to get it to work